# Author: Damyan Ivanov <dmn@debian.org>
# Description: declare inline strings as constants
#  Modern GCC (rightfully) complain otherwise
--- a/src/language_tools.h
+++ b/src/language_tools.h
@@ -27,19 +27,19 @@
 #define LT_(x)	true == lang ? Glib::convert(x[1], "UTF-8", "CP1251"): x[0]
 
 
-static char *HELP_MESSAGE[] = {
+static const char *HELP_MESSAGE[] = {
 	"Type a word (bulgarian or english) in entry box above",
 	"Напишете дума (на български или английски) в полето по-горе"
 };
 
 
-static char *WELLCOME_MESSAGE[] = {
+static const char *WELLCOME_MESSAGE[] = {
 	"Wellcome to GTK BG Office!",
 	"Добре дошли в GTK БГ Офис. Приятно използване!"
 };
 
 
-static char *ABOUT_MESSAGE[] = {
+static const char *ABOUT_MESSAGE[] = {
 	"GTK BG Office assistant - version 1.4 \n"
 	"Official webpage - http://gbgoffice.info\n\n"
 	"(C) 2004-2006 Miroslav Yordanov <mironcho@linux-bg.org>\n"
@@ -52,7 +52,7 @@ static char *ABOUT_MESSAGE[] = {
 };
 
 	
-static char *CONFIG_ERROR[] = {
+static const char *CONFIG_ERROR[] = {
 	"The configuration could not be initialized\n"
 	"This is fatal error and gbgoffice now will exit!",
 	
@@ -61,7 +61,7 @@ static char *CONFIG_ERROR[] = {
 };
 
 
-static char *ERROR_INIT_TRAYICON[] = {
+static const char *ERROR_INIT_TRAYICON[] = {
 	"Error initializing trayicon module.\n"
 	"This is fatal error and gbgoffice now will exit!",
 	
@@ -70,7 +70,7 @@ static char *ERROR_INIT_TRAYICON[] = {
 };
 
 
-static char *DATA_MISSING[] = {
+static const char *DATA_MISSING[] = {
 	"Dicionary files are missing.\n"
 	"Please check that you have installed them\n"
 	"and if they are missing, visit\n"
@@ -86,7 +86,7 @@ static char *DATA_MISSING[] = {
 };
 
 
-static char *DATA_MISSING_FEDORA[] = {
+static const char *DATA_MISSING_FEDORA[] = {
 	"Dicionary files are missing.\n"
 	"Please check that you have installed them\n"
 	"and if they are missing, please use the supplied \n"
@@ -105,190 +105,190 @@ static char *DATA_MISSING_FEDORA[] = {
 
 
 
-static char *GUI_CURRENT_DICT[] = {
+static const char *GUI_CURRENT_DICT[] = {
 	"Current dictionary - ",
 	"Текущ речник - "
 };
 
 
-static char *GUI_NEXT_WORDS[] = {
+static const char *GUI_NEXT_WORDS[] = {
 	"next words",
 	"следващи думи"
 };
 
 
-static char *GUI_MENU_FILE[] = {
+static const char *GUI_MENU_FILE[] = {
 	"_File",
 	"_Файл"
 };
 
 
-static char *GUI_MENU_EDIT[] = {
+static const char *GUI_MENU_EDIT[] = {
 	"_Edit",
 	"Р_едактиране"
 };
 
 
-static char *GUI_MENU_DICTS[] = {
+static const char *GUI_MENU_DICTS[] = {
 	"_Dictionaries",
 	"_Речници"
 };
 
 
-static char *GUI_MENU_SETTINGS[] = {
+static const char *GUI_MENU_SETTINGS[] = {
 	"_Settings",
 	"_Настройки"
 };
 
 
-static char *GUI_MENU_HELP[] = {
+static const char *GUI_MENU_HELP[] = {
 	"_Help",
 	"_Помощ"
 };
 
 
-static char *GUI_VIEW_HISTORY[] = {
+static const char *GUI_VIEW_HISTORY[] = {
 	"View history",
 	"Показва историята"
 };
 
 
-static char *GUI_PREFS_NUM_WORDS[] = {
+static const char *GUI_PREFS_NUM_WORDS[] = {
 	" Number of words in list",
 	" Брой думи в списъка"
 };
 
 
-static char *GUI_PREFS_USE_CLIPBOARD[] = {
+static const char *GUI_PREFS_USE_CLIPBOARD[] = {
 	" Watch clipboard for new words",
 	" Наблюдавай клипборда за нови думи"
 };
 
 
-static char *GUI_PREFS_TAB_GENERAL[] = {
+static const char *GUI_PREFS_TAB_GENERAL[] = {
 	"General",
 	"Основни"
 };
 
 
-static char *GUI_PREFS_TAB_TRAY[] = {
+static const char *GUI_PREFS_TAB_TRAY[] = {
 	"Trayicon",
 	"Trayicon"
 };
 
 
-static char *GUI_PREFS_TAB_TRAY_HELP[] = {
+static const char *GUI_PREFS_TAB_TRAY_HELP[] = {
 	"<b>You must restart gbgoffice \nbefore these settings take effect</b>",
 	"<b>Трябва да рестартирате gbgoffice \nза да влязат в сила тези настройки</b>"
 };
 
 
-static char *GUI_PREFS_USE_TRAYICON[] = {
+static const char *GUI_PREFS_USE_TRAYICON[] = {
 	" Use trayicon",
 	" Използва trayicon"
 };
 
 
-static char *GUI_PREFS_USE_TRAYICON_CLOSE[] = {
+static const char *GUI_PREFS_USE_TRAYICON_CLOSE[] = {
 	" Closing main window,\n quits application",
 	" Затварянето на основния прозорец,\n спира програмата"
 };
 
-static char *GUI_PREFS_TRAYICON_HIDE_ON_START[] = {
+static const char *GUI_PREFS_TRAYICON_HIDE_ON_START[] = {
 	" Hide main window on startup",
 	" Скрива основния прозорец при стартиране"
 };
 
 
-static char *GUI_PREFS_USE_WH[] = {
+static const char *GUI_PREFS_USE_WH[] = {
 	" Use helper",
 	" Използва помощника"
 };
 
 
-static char *GUI_PREFS_WH_SECONDS[] = {
+static const char *GUI_PREFS_WH_SECONDS[] = {
 	" time for showing helper\n (in seconds)",
 	" време за показване на помощника\n (в секунди)"
 };
 
 
-static char *GUI_EXAM_MENU[] = {
+static const char *GUI_EXAM_MENU[] = {
 	"Make a test",
 	"Проверка на знанията"
 };
 
 
-static char *GUI_EXAM_CORRECT[] = {
+static const char *GUI_EXAM_CORRECT[] = {
 	"correct",
 	"правилно"
 };
 
-static char *GUI_EXAM_INCORRECT[] = {
+static const char *GUI_EXAM_INCORRECT[] = {
 	"incorrect",
 	"грешно"
 };
 
-static char *GUI_EXAM_NEWTEST[] = {
+static const char *GUI_EXAM_NEWTEST[] = {
 	"Press button \"New\" for new test.",
 	"Натиснете бутона \"Нов\" за нов тест."
 };
 
-static char *GUI_EXAM_NEW_LEVEL1[] = {
+static const char *GUI_EXAM_NEW_LEVEL1[] = {
 	"Novice",
 	"Начално"
 };
 
-static char *GUI_EXAM_NEW_LEVEL2[] = {
+static const char *GUI_EXAM_NEW_LEVEL2[] = {
 	"Beginner",
 	"Лесно"
 };
 
-static char *GUI_EXAM_NEW_LEVEL3[] = {
+static const char *GUI_EXAM_NEW_LEVEL3[] = {
 	"Intermediate",
 	"Средно"
 };
 
-static char *GUI_EXAM_NEW_LEVEL4[] = {
+static const char *GUI_EXAM_NEW_LEVEL4[] = {
 	"Specialist",
 	"Трудно"
 };
 
-static char *GUI_EXAM_NEW_LEVEL5[] = {
+static const char *GUI_EXAM_NEW_LEVEL5[] = {
 	"Expert",
 	"Експертно"
 };
 
-static char *GUI_EXAM_TRANSLATION[] = {
+static const char *GUI_EXAM_TRANSLATION[] = {
 	"Translation: ",
 	"Превод: "
 };
 
-static char *GUI_EXAM_DIFFICULTY[] = {
+static const char *GUI_EXAM_DIFFICULTY[] = {
 	"Difficulty: ",
 	"Ниво: "
 };
 
-static char *GUI_EXAM_NUMTEST[] = {
+static const char *GUI_EXAM_NUMTEST[] = {
 	"Test (0 = random): ",
 	"Тест (0 = произволен): "
 };
 
-static char *GUI_EXAM_NUMQUEST[] = {
+static const char *GUI_EXAM_NUMQUEST[] = {
 	"Num of questions: ",
 	"Брой въпроси: "
 };
 
-static char *GUI_EXAM_ENDOFTEST[] = {
+static const char *GUI_EXAM_ENDOFTEST[] = {
 	"End of test.",
 	"Край на теста."
 };
 
-static char *GUI_EXAM_TESTNOTSTARTED[] = {
+static const char *GUI_EXAM_TESTNOTSTARTED[] = {
 	"Test not started.",
 	"Не е започнат тест."
 };
 
-static char *GUI_EXAM_CORRECT_ANSWERS[] = {
+static const char *GUI_EXAM_CORRECT_ANSWERS[] = {
 	"correct",
 	"правилни"
 };
